server/hotrod/src/main/java/org/infinispan/server/hotrod/CacheRequestProcessor.java

```diff
@@ -234,8 +234,15 @@ void put(HotRodHeader header, Subject subject, byte[] key, byte[] value, Metadat
 
    private void putInternal(HotRodHeader header, AdvancedCache<byte[], byte[]> cache, byte[] key, byte[] value,
                             Metadata metadata, InfinispanSpan<CacheEntry<?, ?>> span) {
-      cache.putAsyncEntry(key, value, metadata)
-            .whenComplete((ce, throwable) -> handlePut(header, ce, throwable))
+      CompletionStage<CacheEntry<byte[], byte[]>> cs;
+      if (header.hasFlag(ProtocolFlag.ForceReturnPreviousValue)) {
+         cs = cache.putAsyncEntry(key, value, metadata);
+      } else {
+         cs = cache.withFlags(Flag.IGNORE_RETURN_VALUES)
+               .putAsync(key, value, metadata)
+               .thenApply(CompletableFutures.toNullFunction());
```

Comment:
This should always be null as well

Reply:
The `putAsyncEntry` causes an additional synchronized entry clone/creation on the CallInterceptor.

https://github.com/infinispan/infinispan/blob/e8e0d45c869b2f8e22683e9ad09193163383b1c6/core/src/main/java/org/infinispan/interceptors/impl/CallInterceptor.java#L1156-L1161

I saw a few samples on the flame graph and decided to skip the method if it was unneeded.

Reply:
Okay, I would suggest testing the two changes separately at least. Flag.IGNORE_RETURN_VALUES should be a much bigger deal as it avoids sending back the entire value payload for intra cluster communication on the network and an additional unmarshalling.